src/app/submission/submission-edit/subm-edit.component.ts

```diff
@@ -172,14 +174,16 @@ export class SubmEditComponent implements OnInit, OnDestroy, AfterViewChecked {
     }
 
     onSectionDeleteClick(sectionForm: SectionForm): void {
-        let confirmMsg: string = `You are about to permanently delete the page named "${sectionForm.typeName}"`;
+        let confirmMsg = `You are about to permanently delete the page named "${sectionForm.typeName}"`;
 
         if (sectionForm.accno) {
             confirmMsg += ` with accession number ${sectionForm.accno}`;
         }
         confirmMsg += '. This operation cannot be undone.';
 
-        this.confirmPageDelete(confirmMsg).subscribe(() => {
+        this.confirmPageDelete(confirmMsg)
+            .filter(v => v === true)
```

Comment:
filter(...) can go to to the confirmPageDelete(..)

Reply:
or just add another method to the ModalService:
```
whenConfirmed(text: string, title: string, confirmLabel: string): Observable<boolean> {
        return this.confirm(text, title, confirmLabel).filter(v => v === true);
}
```

and use it instead... otherwise you ll have to filter/check true value every time.